### PR TITLE
Add test checking Initial and Off are handled correctly

### PR DESCRIPTION
**Pull request by bskiba**
Extract some common functionality

### CODE REVIEW

File: vertical-pod-autoscaler/e2e/actuation.go [hunk 1]
```diff
@@ -39,54 +39,80 @@ var _ = ActuationSuiteE2eDescribe("Actuation", func() {
...
-		vpaCRD := NewVPA(f, "hamster-vpa", &metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				"app": "hamster",
-			},
-		})
+		ginkgo.By("Setting up a VPA CRD in mode Off")
+		SetupVPA(f, "200m", vpa_types.UpdateModeOff)
 
-		resourceList := apiv1.ResourceList{
-			apiv1.ResourceCPU: ParseQuantityOrDie("9999"), // Request 9999 CPUs to make POD pending
```
Comment (kgolab) [inline vertical-pod-autoscaler/e2e/actuation.go]: I already miss this comment in the new code (wondered for a while what 9999 stands for).

File: vertical-pod-autoscaler/e2e/actuation.go [hunk 2]
```diff
@@ -39,54 +39,80 @@ var _ = ActuationSuiteE2eDescribe("Actuation", func() {
...
+		updatedCPURequest := ParseQuantityOrDie("200m")
+
+		ginkgo.By(fmt.Sprintf("Waiting for pods to be restarted, hoping it won't happen, sleep for %s", VpaEvictionTimeout.String()))
+		CheckNoPodsRestarted(f, podSet)
+		ginkgo.By("Forcefully killing one pod")
+		killPod(f, podList)
+
+		ginkgo.By("Checking that request was modified after forceful restart")
+		updatedPodList, err := GetHamsterPods(f)
+		foundUpdated := false
```
Comment (kgolab) [inline vertical-pod-autoscaler/e2e/actuation.go]: I think the test would be slightly stronger if foundUpdated was an int and compared against the expected value of 1.

File: vertical-pod-autoscaler/e2e/common.go [hunk 3]
```diff
@@ -143,3 +192,68 @@ func ParseQuantityOrDie(text string) resource.Quantity {
...
+	if pods == nil {
+		return result
+	}
+	for _, p := range pods.Items {
+		result[p.Name] = p.UID
+	}
+	return result
+}
+
+// WaitForPodsRestarted waits until pods are restarted.
```
Comment (kgolab) [inline vertical-pod-autoscaler/e2e/common.go]: Maybe "...until some pods from the list are restarted" to clarify the meaning of podList?

**Comment by kgolab** [comment]
Thank you both for the new tests and the accompanying clean-up.
Just a few nits, otherwise LGTM.

File: vertical-pod-autoscaler/e2e/actuation.go (hunk 2 above)
Comment (bskiba) [reply vertical-pod-autoscaler/e2e/actuation.go]: Done

File: vertical-pod-autoscaler/e2e/common.go (hunk 3 above)
Comment (bskiba) [reply vertical-pod-autoscaler/e2e/common.go]: Done

File: vertical-pod-autoscaler/e2e/actuation.go (hunk 1 above)
Comment (bskiba) [reply vertical-pod-autoscaler/e2e/actuation.go]: Ah, I didn't intend to remove it. Done

**Pull request merged**